app/templates/courses/teacher-class-view.jade

```diff
@@ -60,22 +60,34 @@ block content
           .average-playtime.small-details
             span(data-i18n='teacher.avg_playtime')
             span.spr :
-            span= stats.averagePlaytime
+            if view.students.loaded && view.classroom.sessions && view.classroom.sessions.loaded
+              span= stats.averagePlaytime
+            else
+              span(data-i18n='common.loading')
 
           .total-playtime.small-details
             span(data-i18n='teacher.total_playtime')
             span.spr :
-            span= stats.totalPlaytime
+            if view.students.loaded && view.classroom.sessions && view.classroom.sessions.loaded
+              span= stats.totalPlaytime
+            else
+              span(data-i18n='common.loading')
 
           .average-complete.small-details
             span(data-i18n='teacher.avg_completed')
             span.spr :
-            span= stats.averageLevelsComplete
+            if view.students.loaded && view.classroom.sessions && view.classroom.sessions.loaded
+              span= stats.averageLevelsComplete
+            else
+              span(data-i18n='common.loading')
 
           .total-complete.small-details
             span(data-i18n='teacher.total_completed')
             span.spr :
-            span= stats.totalLevelsComplete
+            if view.students.loaded && view.classroom.sessions && view.classroom.sessions.loaded
```

Comment:
Seems like this should be turned into `view.allStatsLoaded()` or something named similar. Just have to create the function and call it in all four cases.